nameserver/nameserver.go

```diff
@@ -203,11 +203,29 @@ func (n *Nameserver) receiveGossip(msg []byte) (mesh.GossipData, mesh.GossipData
 		return nil, nil, fmt.Errorf("clock skew of %d detected", delta)
 	}
 
+	gossip.Entries.filter(func(e *Entry) bool {
```

Comment:
deserves a comment as to why we are doing this here rather than as part of the filter below.